src/cipher.rs

```diff
@@ -530,14 +533,54 @@ impl rustls_client_cert_verifier {
     /// rustls_client_cert_verifier_free for details about lifetime.
     /// This copies the contents of the rustls_root_cert_store. It does not take
     /// ownership of the pointed-to memory.
+    // NOTE(@cpu): return type changed to *mut because we need to be able to pass the
+    //             results of this constructor, and the with_crls constructor to free and one of
+    //             those returns a mut pointer. Maybe there's a better way?
+    //             A union for the free arg?
     #[no_mangle]
     pub extern "C" fn rustls_client_cert_verifier_new(
         store: *const rustls_root_cert_store,
-    ) -> *const rustls_client_cert_verifier {
+    ) -> *mut rustls_client_cert_verifier {
+        ffi_panic_boundary! {
+            let store: &RootCertStore = try_ref_from_ptr!(store);
+            let client_cert_verifier = AllowAnyAuthenticatedClient::new(store.clone());
+            return Arc::into_raw(client_cert_verifier.boxed()) as *mut _;
+        }
+    }
+
+    /// TODO(@cpu): write docstring..
+    // TODO(@cpu): some notes...:
+    //      * Need this to take a verifier_out *mut *mut because parsing CRLs can error, so we
+    //        have to use up the return type w/ a rustls_result.
+    //      * As written, only supports a single CRL. Need to think about how to express multiple
+    //        *const u8 with different size_t lengths...
+    //      * This isn't wrapping anything in an arc - I think what we want is an Arc<Box<_>> but I
+    //        can't figure out how to express that with the existing macros. `set_mut_ptr`'s second
+    //        arg must be a AllowAnyAuthenticatedClient :-/
+    #[no_mangle]
+    pub extern "C" fn rustls_client_cert_verifier_new_with_crls(
+        verifier_out: *mut *mut rustls_client_cert_verifier,
```

Comment:
I thought maybe this could be a `*mut *const` since we want to update the pointer to the pointer to the verifier but not the pointer to the verifier itself but I can't express this with the existing macros. `set_mut_ptr` requires *mut mut